quantgov/corpora/builtins.py

```diff
@@ -93,3 +101,100 @@ def process_document(doc, terms, pattern, total_label):
 
 
 commands['count_occurrences'] = OccurrenceCounter
+
+
+class ShannonEntropy():
+
+    cli = quantgov.utils.CLISpec(
+        help='Shannon Entropy',
+        arguments=[
+            quantgov.utils.CLIArg(
+                flags=('--word_pattern', '-wp'),
+                kwargs={
+                    'help': 'regular expression defining a "word"',
+                    'type': re.compile,
+                    'default': re.compile(r'\b\w+\b')
+                }
+            ),
+            quantgov.utils.CLIArg(
+                flags=('--stopwords', '-sw'),
+                kwargs={
+                    'help': 'stopwords to ignore',
+                    'default': stopwords.words('english')
+                }
+            )
+        ]
+
+    )
+
+    @staticmethod
+    def get_columns(args):
+        return ('shannon_entropy',)
+
+    @staticmethod
+    def process_document(doc, word_pattern, stopwords):
+        words = word_pattern.findall(doc.text)
+        lemmas = [
+            lemma for lemma in (
+                Word(word).lemmatize() for word in words
+            )
+            if lemma not in stopwords
+        ]
+        counts = collections.Counter(lemmas)
+        return doc.index + (round(sum(
+            -(count / len(lemmas) * math.log(count / len(lemmas), 2))
+            for count in counts.values()
+        ), 2),)
+
+
+commands['shannon_entropy'] = ShannonEntropy
+
+
+class ConditionalCounter():
+
+    cli = quantgov.utils.CLISpec(
+        help='Conditional Counter',
+        arguments=[
+            quantgov.utils.CLIArg(
+                flags=('--pattern'),
+                kwargs={
```

Comment:
I don't think we really need this option. In fact, we could even take this as a special case of "count occurences", and just wrap that function, wouldn't we?